UIforETW/Utility.cpp

```diff
@@ -19,7 +19,166 @@ limitations under the License.
 #include <fstream>
 #include <direct.h>
 
-std::vector<std::wstring> split(const std::wstring& s, char c)
+
+namespace {
+PCWSTR const WPAStartupFileName = L"\\Startup.wpaProfile";
+
+std::wstring GetDocumentsFolderPath()
+{
+	//must CoTaskMemFree when done!
+	PWSTR docsPathTemp = NULL;
+	const HRESULT docsPathResult = ::SHGetKnownFolderPath(FOLDERID_Documents, KF_FLAG_NO_ALIAS, NULL, &docsPathTemp);
+	if (FAILED(docsPathResult))
+	{
+		if (docsPathResult == E_INVALIDARG)

```

Comment:
How does the error handling here add value? If we can't get the documents folder path then can we really continue? Why not delete 15 lines of code and just terminate if the docs folder can't be found? If the error handling adds value I want to know what it is.